Havtorn/Source/Engine/Graphics/RenderManager.cpp

```diff
@@ -813,6 +787,239 @@ namespace Havtorn
+		depthStencilViewDesc.Format = DXGI_FORMAT_D32_FLOAT;
+		depthStencilViewDesc.ViewDimension = D3D11_DSV_DIMENSION_TEXTURE2D;
+		depthStencilViewDesc.Texture2D.MipSlice = 0;
+		depthStencilViewDesc.Flags = 0;
+
+		ID3D11Texture2D* depthStencilBuffer;
+		ENGINE_HR_MESSAGE(Framework->GetDevice()->CreateTexture2D(&depthStencilDesc, nullptr, &depthStencilBuffer), "Texture could not be created.");
+
+		ID3D11DepthStencilView* depthStencilView;
+		ENGINE_HR_MESSAGE(Framework->GetDevice()->CreateDepthStencilView(depthStencilBuffer, &depthStencilViewDesc, &depthStencilView), "Depth could not be created.");
+
+		ID3D11Texture2D* texture;
+		ENGINE_HR_MESSAGE(Framework->GetDevice()->CreateTexture2D(&textureDesc, nullptr, &texture), "Could not create Fullscreen Texture2D");
+
+		ID3D11ShaderResourceView* shaderResource;
+		ENGINE_HR_MESSAGE(Framework->GetDevice()->CreateShaderResourceView(texture, nullptr, &shaderResource), "Could not create Fullscreen Shader Resource View.");
+
+		ID3D11RenderTargetView* renderTarget;
+		ENGINE_HR_MESSAGE(Framework->GetDevice()->CreateRenderTargetView(texture, nullptr, &renderTarget), "Could not create Fullcreen Render Target View.");
+
+		D3D11_VIEWPORT* viewport;
+		D3D11_TEXTURE2D_DESC textureDescription;
+		texture->GetDesc(&textureDescription);
+		viewport = new D3D11_VIEWPORT({ 0.0f, 0.0f, static_cast<F32>(textureDescription.Width), static_cast<F32>(textureDescription.Height), 0.0f, 1.0f });
+
+		CGBuffer gBuffer = FullscreenTextureFactory.CreateGBuffer({ static_cast<F32>(textureDescription.Width), static_cast<F32>(textureDescription.Height) });
+		gBuffer.SetAsActiveTarget();
+		Context->RSSetViewports(1, viewport);
+
+		STransform camTransform;
+		constexpr F32 zoomMultiplier = 0.72f;
+		camTransform.Translate(SVector4::Backward * 1.8f * zoomMultiplier);
+		camTransform.Translate(SVector4::Right * 1.08f * zoomMultiplier);
+		camTransform.Translate(SVector4::Up * 1.2f * zoomMultiplier);
+		camTransform.Rotate({ UMath::DegToRad(-30.0f), UMath::DegToRad(30.0f), 0.0f });
+		SMatrix camProjection = SMatrix::PerspectiveFovLH(UMath::DegToRad(70.0f), 1.0f, 0.01f, 10.0f);
+
+		FrameBufferData.ToCameraFromWorld = camTransform.GetMatrix().FastInverse();
+		FrameBufferData.ToWorldFromCamera = camTransform.GetMatrix();
+		FrameBufferData.ToProjectionFromCamera = camProjection;
+		FrameBufferData.ToCameraFromProjection = camProjection.Inverse();
+		FrameBufferData.CameraPosition = camTransform.GetMatrix().GetTranslation4();
+		BindBuffer(FrameBuffer, FrameBufferData, "Frame Buffer");
+
+		Context->VSSetConstantBuffers(0, 1, &FrameBuffer);
+		Context->PSSetConstantBuffers(0, 1, &FrameBuffer);
+
+		ObjectBufferData.ToWorldFromObject = SMatrix();
+		BindBuffer(ObjectBuffer, ObjectBufferData, "Object Buffer");
+
+		Ref<SEntity> tempEntity = std::make_shared<SEntity>(0, "Temp");
+		SMaterialComponent* materialComp = new SMaterialComponent(tempEntity, EComponentType::MaterialComponent);
```

Comment:
materialComp, environmentLightComp and directionalLightComp are allocated but not destroyed (can't see where this is done)? Smart pointer is used for tempEntity but for the components, risk of memory leak as Entity is not responsible for destroying its components?